addons/hr_recruitment/wizard/applicant_refuse_reason.py

```diff
@@ -1,14 +1,114 @@
 # -*- coding: utf-8 -*-
+import logging
 
-from odoo import api, fields, models
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
 
 
 class ApplicantGetRefuseReason(models.TransientModel):
     _name = 'applicant.get.refuse.reason'
     _description = 'Get Refuse Reason'
 
+    @api.model
+    def default_get(self, fields):
+        if not self.env.user.email:
+            raise UserError(_("Unable to post message, please configure the sender's email address."))
+        res = super().default_get(fields)
+        applicant_ids = self.env.context.get('default_applicant_ids')
+        if 'partner_ids' in fields and applicant_ids:
+            res.update({'partner_ids': self.env['hr.applicant'].browse(applicant_ids).mapped('partner_id.id')})
+        return res
+
     refuse_reason_id = fields.Many2one('hr.applicant.refuse.reason', 'Refuse Reason')
     applicant_ids = fields.Many2many('hr.applicant')
+    partner_ids = fields.Many2many('res.partner', string='Recipients')
+    applicant_without_email = fields.Text(compute='_compute_applicant_without_email', string='applicant(s) not having email')
+    subject = fields.Char('Subject', compute='_compute_body', store=True, readonly=False)
+    body = fields.Html('Contents', sanitize_style=True, compute='_compute_body', store=True, readonly=False)
+    attachment_ids = fields.Many2many(
+        'ir.attachment', 'hr_recruitment_mail_compose_message_ir_attachments_rel',
+        'wizard_id', 'attachment_id', string='Attachments')
+    template_id = fields.Many2one('mail.template', "Email Templates", compute='_compute_template_id', store=True, readonly=False, domain="[('model', '=', 'hr.applicant')]")
+    email_from = fields.Char(
+        'From', required=True,
+        default=lambda self: self.env.user.email_formatted,
+        help="Email address of the sender",
+    )
+    author_id = fields.Many2one(
+        'res.partner', string='Author', required=True,
+        default=lambda self: self.env.user.partner_id.id,
+        help="Author of the message.",
+    )
+
+    @api.depends('applicant_ids')
+    def _compute_applicant_without_email(self):
+        for wizard in self:
+            applicants = self.env['hr.applicant'].search([
+                ('id', 'in', wizard.applicant_ids.ids),
+                '|', ('email_from', '=', False), ('partner_id.email', '=', False)
+            ])
+            if applicants:
+                wizard.applicant_without_email = "%s\n%s" % (
+                    _("The email will not be sent to following applicant(s) as they don't have email address."),
+                    "\n".join([i.partner_name for i in applicants])
+                )
+            else:
+                wizard.applicant_without_email = False
+
+    @api.depends('refuse_reason_id')
+    def _compute_template_id(self):
+        for wizard in self.filtered(lambda x: x.refuse_reason_id):
```

Comment:
You should also assign a value for the records without `refuse_reason_id`

```suggestion
        for wizard in self.filtered('refuse_reason_id'):
```